syncd/SaiSwitch.cpp

```diff
@@ -131,6 +137,26 @@ void SaiSwitch::getDefaultMacAddress(
     memcpy(mac, m_default_mac_address, sizeof(sai_mac_t));
 }
 
+sai_switch_type_t SaiSwitch::getSwitchType() const
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attr;
+
+    attr.id = SAI_SWITCH_ATTR_TYPE;
+
+    sai_status_t status = m_vendorSai->get(SAI_OBJECT_TYPE_SWITCH, m_switch_rid, 1, &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("failed to get switch type");
```

Comment:
in case of failure we should throw here, or return explicitly default switch type

Reply:
Ok, will throw here